pyriemann/clustering.py

```diff
@@ -483,11 +489,231 @@ def _check_labels(self, X, y):
+            Labels corresponding to each trial. If None, all trials are
+            considered as clean.
+
+        Returns
+        -------
+        self : Potato field instance
+            The Potato field instance.
+        """
+        if self.n_potatoes < 1:
+            raise ValueError('Parameter n_potatoes must be at least 1')
+        if not 0 < self.p_threshold < 1:
+            raise ValueError('Parameter p_threshold must be in (0, 1)')
+        self._check_length(X)
+        n_trials = X[0].shape[0]
+
+        pt = Potato(
+            metric=self.metric,
+            threshold=self.z_threshold,
+            n_iter_max=self.n_iter_max,
+            pos_label=self.pos_label,
+            neg_label=self.neg_label)
+        self._potatoes = []
+        for i in range(self.n_potatoes):
+            if X[i].shape[0] != n_trials:
+                raise ValueError(
+                    'Unequal n_trials between ndarray of X. Should be %d but '
+                    'got %d.' % (n_trials, X[i].shape[0]))
+            self._potatoes.append(clone(pt))
+            self._potatoes[i].fit(X[i], y)
+
+        return self
+
+    def partial_fit(self, X, y=None, alpha=0.1):
+        """Partially fit the potato field from covariance matrices.
+
+        This partial fit can be used to update dynamic or semi-dymanic online
+        potatoes with clean EEG [1]_.
+
+        Parameters
+        ----------
+        X : list of n_potatoes ndarray of shape (n_trials, n_channels,
+                n_channels), with same n_trials but different n_channels
+            list of ndarray of SPD matrices.
+        y : ndarray, shape (n_trials,) | None (default None)
+            Labels corresponding to each trial. If None, all trials are
+            considered as clean.
+        alpha : float (default 0.1)
+            Update rate in [0, 1] for the centroid, and mean and standard
+            deviation of log-distances: 0 for no update, 1 for full update.
+
+        Returns
+        -------
+        self : Potato instance
+            The Potato instance.
+        """
+        self._check_length(X)
+        n_trials = X[0].shape[0]
+
+        for i in range(self.n_potatoes):
+            if X[i].shape[0] != n_trials:
+                raise ValueError(
+                    'Unequal n_trials between ndarray of X. Should be %d but '
+                    'got %d.' % (n_trials, X[i].shape[0]))
+            self._potatoes[i].partial_fit(X[i], y, alpha=alpha)
+        return self
+
+    def transform(self, X):
+        """Return the normalized log-distances to the centroids (ie geometric
+        z-scores of distances).
+
+        Parameters
+        ----------
+        X : list of n_potatoes ndarray of shape (n_trials, n_channels,
+                n_channels), with same n_trials but different n_channels
+            list of ndarray of SPD matrices.
+
+        Returns
+        -------
+        z : ndarray, shape (n_potatoes, n_trials)
```

Comment:
```suggestion
        z : ndarray, shape (n_trials, n_potatoes)
```
would be more natural